ST with secrets - mysql, elasticsearch, postgresql. Customer facing doc for setting secrets as env variable (#4945)
SumoLogic/sumologic-documentation: docs/send-data/opentelemetry-collector/remote-management/source-templates/postgresql/index.md

---
slug: /send-data/opentelemetry-collector/remote-management/source-templates/postgresql
title: PostgreSQL Source Template
sidebar_label: PostgreSQL
description: Learn about the Sumo Logic PostgreSQL source template for OpenTelemetry.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<head> 
  <meta name="robots" content="noindex" />

> </head>

Reviewer: Since this doc is present in the sidebars, I'd remove this code snippet, which is only intended for beta and/or hidden docs. It blocks it from Algolia and Google search.

Reviewer: @sumoanema

Author: Done. Thanks for pointing that out @kimsauce

## Fields creation in Sumo Logic for PostgreSQL

If not already present, the following [Fields](/docs/manage/fields/) are created as part of source template creation.

- **`sumo.datasource`**. Fixed value of **postgresql**.
- **`deployment.environment`**. This is a user-configured field set at the time of collector installation. It identifies the environment where the host resides, such as `dev`, `prod`, or `qa`.
- **`db.cluster.name`**. User configured. Enter a uniquely identifiable name for your PostgreSQL cluster to show in the Sumo Logic dashboards.
- **`db.node.name`**. Includes the value of the hostname of the machine which is being monitored.

## Prerequisites

### For metrics collection

- The PostgreSQL metrics [receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/postgresqlreceiver) collects metrics by querying the PostgreSQL [statistics collector](https://www.postgresql.org/docs/13/monitoring-stats.html).
- The monitoring user which is used in the source template must be granted permission to SELECT permission for [pg_stat_database](https://www.postgresql.org/docs/current/monitoring-stats.html#MONITORING-PG-STAT-DATABASE-VIEW).

> - Make sure to set the password of PostgreSQL user as an environment variable for OpenTelemetry agent, refer to the [Setting Environment variable with secret values for Source Template](../../st-with-secrets.md).

Reviewer: ```suggestion
- Make sure to set the PostgreSQL user's password as an environment variable for the OpenTelemetry agent. Refer to [[Setting Environment Variables with Secret Values for Source Templates]](../../st-with-secrets.md).
```

:::note

> For SumoLogic OpenTelemetry collector version below 0.116 metric collection will work only for PostgreSQL version 16 and below. With Sumo Logic OpenTelemetry collector version 0.116 and above metric collection will work for PostgreSQL version 17 as well.

Reviewer: ```suggestion
For Sumo Logic OpenTelemetry Collector versions below 0.116, metric collection supports only PostgreSQL version 16 and earlier. With Sumo Logic OpenTelemetry collector version 0.116 and above, metric collection also supports PostgreSQL version 17.
```
